stock_move_location/wizard/stock_move_location.py

```diff
@@ -0,0 +1,332 @@
+# Copyright (C) 2011 Julius Network Solutions SARL <contact@julius.fr>
+# Copyright 2018 Camptocamp SA
+# Copyright 2019 Sergio Teruel - Tecnativa <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from itertools import groupby
+
+from odoo import api, fields, models
+from odoo.fields import first
+
+
+class StockMoveLocationWizard(models.TransientModel):
+    _name = "wiz.stock.move.location"
+    _description = "Wizard move location"
+
+    def _get_default_picking_type_id(self):
+        company_id = self.env.context.get("company_id") or self.env.user.company_id.id
+        return (
+            self.env["stock.picking.type"]
+            .search(
+                [
+                    ("code", "=", "internal"),
+                    ("warehouse_id.company_id", "=", company_id),
+                ],
+                limit=1,
+            )
+            .id
+        )
+
+    origin_location_disable = fields.Boolean(
+        compute="_compute_readonly_locations",
+        help="technical field to disable the edition of origin location.",
+    )
+    origin_location_id = fields.Many2one(
+        string="Origin Location",
+        comodel_name="stock.location",
+        required=True,
+        domain=lambda self: self._get_locations_domain(),
+    )
+    destination_location_disable = fields.Boolean(
+        compute="_compute_readonly_locations",
+        help="technical field to disable the edition of destination location.",
+    )
+    destination_location_id = fields.Many2one(
+        string="Destination Location",
+        comodel_name="stock.location",
+        required=True,
+        domain=lambda self: self._get_locations_domain(),
+    )
+    stock_move_location_line_ids = fields.Many2many(
+        string="Move Location lines",
+        comodel_name="wiz.stock.move.location.line",
+        column1="move_location_wiz_id",
+        column2="move_location_line_wiz_id",
+    )
+    picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type", default=_get_default_picking_type_id
+    )
+    picking_id = fields.Many2one(
+        string="Connected Picking", comodel_name="stock.picking"
+    )
+    edit_locations = fields.Boolean(default=True)
+    apply_putaway_strategy = fields.Boolean()
+
```

Comment:
```suggestion
`clear_move_lines = fields.Boolean()
@api.onchange("clear_move_lines")
    def clear_move_line(self):
        if self.clear_move_lines == True:
            self._clear_lines()
`
```

Reply:
If the user want to move only some products from source to destination and there is a lot of products in source location, user need to remove products one by one. So, I think we should add the option for removing all line. That can save time.